### PR TITLE
meshctl cmds (cluster deregister, install, uninstall, check, describe mesh, describe meshservice)

### PR DESCRIPTION
**Pull request by harveyxia**
Implements the following `meshctl` commands:

- `cluster`
  - `deregister`
- `install`
- `uninstall`
- `check` (this one is partially done, we still need to implement the connectivity checks to remote clusters)
- `describe`
  - `mesh`
  - `meshservice`

### CODE REVIEW

File: ci/setup-kind.sh [hunk 1]
```diff
@@ -14,15 +14,18 @@
...
   kind get clusters | grep -E "${masterCluster}|${remoteCluster}" | while read -r r; do kind delete cluster --name "${r}"; done
   exit 0
 fi
 
+PROJECT_ROOT=$( cd "$( dirname "${0}" )" >/dev/null 2>&1 && pwd )/..
+echo "Using project root ${PROJECT_ROOT}"
+
+source ${PROJECT_ROOT}/ci/setup-funcs.sh
+
+make clean || true
```
Comment (ilackarms) [inline ci/setup-kind.sh:27]: risky to put this here; i'd put it in the CI step instead (don't want to accidentally blow away local changes when running tests locally)

File: ci/setup-kind.sh [hunk 2]
```diff
@@ -33,19 +36,19 @@ install_istio ${remoteCluster} 32000 &
...
 
 # install service mesh hub
-${PROJECT_ROOT}/ci/setup-smh.sh ${masterCluster}
+install_smh ${masterCluster}
 
 # sleep to allow crds to register
 sleep 4
 
-# register clusters
-register_cluster ${masterCluster} &
```
Comment (ilackarms) [inline ci/setup-kind.sh:45]: why remove registration of the master cluster?

File: install/helm/helm.mk [hunk 3]
```diff
@@ -16,25 +13,25 @@ clean-helm:
 	rm $(CA_CHART_DIR)/values.yamls
 
 .PHONY: package-helm
-package-helm:
-	helm package --destination $(PACKAGED_CHARTS_DIR) $(SMH_CHART_DIR)
-	helm package --destination $(PACKAGED_CHARTS_DIR) $(CA_CHART_DIR)
+package-helm: chart-gen
+	helm package --destination $(CHART_OUTPUT_DIR)/service-mesh-hub $(SMH_CHART_DIR)
+	helm package --destination $(CHART_OUTPUT_DIR)/cert-agent $(CA_CHART_DIR)
```
Comment (ilackarms) [inline install/helm/helm.mk:18]: what was the reasoning behind this change?

File: pkg/meshctl/commands/check/check.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	opts := &options{}
+	cmd := &cobra.Command{
+		Use:   "check",
+		Short: "Perform health checks on the Service Mesh Hub system",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := utils.BuildClient(opts.kubeconfig, opts.kubecontext)
+			if err != nil {
+				return err
+			}
+			return internal.RunChecks(ctx, client, opts.namespace)
```
Comment (ilackarms) [inline pkg/meshctl/commands/check/check.go:23]: not sure about the use of `internal` package here. can we follow the patterns used in other CLI commands?

File: pkg/meshctl/commands/check/internal/checks.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+var (
+	checkMarkChar = "\u2705"
+	redXChar      = "\u274C"
+
+	// TODO implement kube connectivity check
+
+	managementPlane = Category{
+		Name: "Management Plane",
```
Comment (ilackarms) [inline pkg/meshctl/commands/check/internal/checks.go:43]: i'm not sure about the usage of the term `Management Plane` here. we should discuss & converge around some of the terminology we're using for SMH components.

File: ci/setup-kind.sh (hunk 2 above)
Comment (harveyxia) [reply ci/setup-kind.sh:45]: This is being done [as part of `meshctl install`](https://github.com/solo-io/service-mesh-hub/pull/860/files#diff-7212c0ee92938d1d577b8949668fab52R47)

File: install/helm/helm.mk (hunk 3 above)
Comment (harveyxia) [reply install/helm/helm.mk:18]: When we generate the Helm index file using `helm repo index`, all Helm archives in the referenced directory will be included in the index, so we need to create a separate directory for each Helm chart

File: pkg/meshctl/commands/check/internal/checks.go (hunk 5 above)
Comment (harveyxia) [reply pkg/meshctl/commands/check/internal/checks.go]: changing to "Service Mesh Hub" for now

File: pkg/meshctl/commands/check/check.go (hunk 4 above)
Comment (harveyxia) [reply pkg/meshctl/commands/check/check.go]: moved the checks to `pkg/meshctl/checks`

File: pkg/meshctl/checks/networking_crd_check.go [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/rotisserie/eris"
+	"github.com/solo-io/service-mesh-hub/pkg/api/networking.smh.solo.io/v1alpha2"
+	"github.com/solo-io/skv2/pkg/ezkube"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type networkingCrdStatus interface {
```
Comment (ilackarms) [inline pkg/meshctl/checks/networking_crd_check.go:14]: i'd comment the purpose behind this interface here

Comment (ilackarms) [reply pkg/meshctl/checks/networking_crd_check.go:14]: i'd also move it to be right above `checkStatus`

File: pkg/meshctl/commands/cluster/deregister/cluster_deregister.go [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+	set.StringVar(&opts.KubeContext, "master-context", "", "name of the kubeconfig context to use for the master cluster")
+	set.StringVar(&opts.RemoteKubeContext, "remote-context", "", "name of the kubeconfig context to use for the remote cluster")
+	set.StringVar(&opts.Namespace, "federation-namespace", defaults.DefaultPodNamespace, "namespace of the Service-Mesh-Hub control plane in which the secret for the deregistered cluster will be created")
+	set.StringVar(&opts.RemoteNamespace, "remote-namespace", defaults.DefaultPodNamespace, "namespace in the target cluster where a service account enabling remote access will be created. If the namespace does not exist it will be created.")
+	set.StringVar(&opts.APIServerAddress, "api-server-address", "", "Swap out the address of the remote cluster's k8s API server for the value of this flag. Set this flag when the address of the cluster domain used by the Service Mesh Hub is different than that specified in the local kubeconfig.")
+	set.BoolVar(&opts.Verbose, "verbose", true, "enable/disable verbose logging during installation of cert-agent")
```
Comment (ilackarms) [inline pkg/meshctl/commands/cluster/deregister/cluster_deregister.go:34]: as discussed, let's pull these opts out to a shared package with cluster/register in cluster/common

let's also employ the convention of using a typed `Options` which implements the `AddToFlags` method

File: pkg/meshctl/commands/cluster/register/cluster_register.go [hunk 8]
```diff
@@ -3,103 +3,35 @@ package register
...
-		HelmChartOverride: opts.chartPath,
-		HelmValuesPath:    opts.chartValues,
-		KubeConfig:        opts.RemoteKubeCfgPath,
-		KubeContext:       opts.RemoteKubeContext,
-		Namespace:         opts.RemoteNamespace,
-		Verbose:           false,
-	}.InstallCertAgent(
-		ctx,
-	)
+func addToFlags(opts *registration.RegistrantOptions, set *pflag.FlagSet) {
```
Comment (ilackarms) [inline pkg/meshctl/commands/cluster/register/cluster_register.go:25]: let's employ the convention of using a typed `Options` which implements the `AddToFlags` method

File: pkg/meshctl/checks/networking_crd_check.go (hunk 6 above)
Comment (harveyxia) [reply pkg/meshctl/checks/networking_crd_check.go:14]: 👍

File: pkg/meshctl/commands/cluster/deregister/cluster_deregister.go (hunk 7 above)
Comment (harveyxia) [reply pkg/meshctl/commands/cluster/deregister/cluster_deregister.go:34]: discussed offline, there are slight copy differences between `register` and `deregister` that make it awkward to hoist into common flags

**Pull request merged**